src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -48,7 +48,7 @@ public function check($value = null, $schema = null, $path = null, $i = null)
             $validatedOneType = false;
             $errors = array();
             foreach ($type as $tp) {
-                $validator = new static($this->checkMode);
+                $validator = $this->getFactory()->createInstanceFor('type');
```

Comment:
@FlorianSW parent constraint passes `checkMode` to the factory which in turn passes `checkMode` to the new constraint, so i don't see any problems here